lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3OutputFile.java

```diff
@@ -45,15 +46,33 @@ public S3OutputFile(Executor uploadExecutor, S3Client client, S3Context context,
     public void createOrOverwrite(byte[] data)
             throws IOException
     {
-        try (OutputStream out = create()) {
+        try (OutputStream out = create(newSimpleAggregatedMemoryContext(), false)) {
+            out.write(data);
+        }
+    }
+
+    @Override
+    public void createExclusive(byte[] data)
+            throws IOException
+    {
+        if (!context.exclusiveWriteSupported()) {
+            throw new UnsupportedOperationException("createExclusive not supported by " + getClass());
+        }
+
+        try (OutputStream out = create(newSimpleAggregatedMemoryContext(), true)) {
             out.write(data);
         }
     }
 
     @Override
     public OutputStream create(AggregatedMemoryContext memoryContext)
     {
-        return new S3OutputStream(memoryContext, uploadExecutor, client, context, location);
+        return create(memoryContext, context.exclusiveWriteSupported());
```

Comment:
does going with exclusive mode by default can have some negative impact. E.g performance wise?

Reply:
I don't think so. According to the docs, the multipart close does the "not exists" check. Also: `This feature is available at no additional charge in all AWS Regions, including the AWS GovCloud (US) Regions and the AWS China Regions.`